identity-service/src/utils/apiSigning.js

```diff
@@ -0,0 +1,67 @@
+const Web3 = require('web3')
+const web3 = new Web3()
+
+// TODO: This is copied from the same code path in content node
+// and should be standardized across this file as well as the method in libs
+
+/**
+ * Max age of signature in milliseconds
+ * Set to 5 minutes
+ */
+const MAX_SIGNATURE_AGE_MS = 300000
+
+/**
+ * Generate the timestamp and signature for api signing
+ * @param {object} data
+ * @param {string} privateKey
+ */
+const generateTimestampAndSignature = (data, privateKey) => {
+  const timestamp = new Date().toISOString()
+  const toSignObj = { ...data, timestamp }
+  // JSON stringify automatically removes white space given 1 param
+  const toSignStr = JSON.stringify(sortKeys(toSignObj))
+  const toSignHash = web3.utils.keccak256(toSignStr)
+  const signedResponse = web3.eth.accounts.sign(toSignHash, privateKey)
```

Comment:
Can this guy throw? Do we want to guard for that?

Reply:
don't want this to drift from code in CN, but we should look at separately